server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -4126,7 +4162,7 @@ protected void ensureMaxSeqNoEqualsToGlobalCheckpoint(SeqNoStats seqNoStats) {
         closeShards(readonlyShard);
     }
 
-    public void testCloseShardWhileEngineIsWarming() throws Exception {
+    public void testEngineWarmingDoesNotBlockClusterUpdate() throws Exception {
```

Comment:
I think we want both cluster state updating and closing to be possible while warming?